src/Faker/Provider/en_GB/Company.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Faker\Provider\en_GB;
+
+class Company extends \Faker\Provider\Company
+{
+    public const VAT_PREFIX = 'GB';
+    public const VAT_TYPE_DEFAULT = 'vat';
+    public const VAT_TYPE_BRANCH = 'branch';
+    public const VAT_TYPE_GOVERNMENT = 'gov';
+    public const VAT_TYPE_HEALTH_AUTHORITY = 'health';
+
+    /**
+     * UK VAT number
+     *
+     * @see https://en.wikipedia.org/wiki/VAT_identification_number#VAT_numbers_by_country
+     */
+    public static function vat(string $type = null): string
+    {
+        switch ($type) {
+            /**
+             * branch traders:
+             * 12 digits (as for 9 digits, followed by a block of 3 digits)
+             */
+            case static::VAT_TYPE_BRANCH:
+                return sprintf(
+                    '%s%d %d %d %d',
+                    static::VAT_PREFIX,
+                    static::randomNumber(3, true),
+                    static::randomNumber(4, true),
+                    static::randomNumber(2, true),
+                    static::randomNumber(3, true)
+                );
+            /**
+             * government departments:
+             * the letters GD then 3 digits from 000 to 499 (e.g. GBGD001)
+             */
+            case static::VAT_TYPE_GOVERNMENT:
+                return sprintf(
+                    '%sGD%d',
+                    static::VAT_PREFIX,
+                    str_pad((string) static::numberBetween(0, 499), 3)
```

Comment:
There's a bug here that can be reproduced by adding `$this->faker->seed(4);` to `testVatGovernmentType()`: this produces `GBGD17` (which does not have the expected format `GBGD017`).

To solve this replace the `sprintf` format and `str_pad`:
```suggestion
                    '%sGD%s',
                    static::VAT_PREFIX,
                    str_pad(static::numberBetween(0, 499), 3, '0', STR_PAD_LEFT)
```